libultraship/libultraship/Audio.cpp

```diff
@@ -24,15 +24,15 @@ namespace Ship
 
 		int loopStateCnt = reader->ReadUInt32();
 
-		for (uint32_t i = 0; i < loopStateCnt; i++)
+		for (int i = 0; i < loopStateCnt; i++)
 			entry->loop.states.push_back(reader->ReadInt16());
 
 		entry->book.order = reader->ReadInt32();
 		entry->book.npredictors = reader->ReadInt32();
 
 		int bookSize = reader->ReadInt32();
 
-		for (uint32_t i = 0; i < bookSize; i++)
+		for (int i = 0; i < bookSize; i++)
```

Comment:
```suggestion
		for (size_t i = 0; i < bookSize; i++)
```

Reply:
so book size is coming from `int bookSize = reader->ReadInt32()` which is already saving it to an `int`. Comparing it here as `uint32_t` makes the type checker think you're comparing two wrong types. I'll update the places where we do `ReadInt32();` to actually store as `uint32_t` since that's what it returns.

Reply:
So size_t isn't just a size. It's not just an unsigned integer either. It's actually just meant as a _size in bytes._ The loop counters here appear to be reading in bytes based on the value, thus size_t fits.

It may be this is inconsistent and more places need to become size_t as well.